src/cocaine-app/yt_worker.py

```diff
@@ -0,0 +1,176 @@
+        result = None
+
+        with YqlClient(db=self._cluster, token=self._token) as yql:
+            try:
+                if not yql.ping():
+                    raise IOError("YQL ping failed {} {}".format(self._cluster, self._token))
+            except:
+                logger.exception("YQL ping excepted {}{}".format(self._cluster, self._token))
+                raise
+
+            start_time = time.time()
+
+            for attempt in xrange(self._attempts):
+                try:
+                    request = yql.query(query)
+                    request.run()
+
+                    logger.info("YQL query is running with id {}".format(request.operation_id))
+                    result = request.results
+                    end_time = time.time()
+                    if result.is_success:
+                        logger.info("YQL query succeeded and took {}".format(end_time-start_time))
+                        if timeout and (end_time - start_time > timeout):
+                            raise AssertionError("YQL query({}..{}) > timeout {}".format(start_time, end_time, timeout))
+                        return result
+
+                    logger.error("YQL failed #{}: {}[{}] ({},{!s})".format(attempt, result.status,
+                                 request.status_code, request.explain(), request.exc_info))
+                    for error in result.errors:
+                        logger.error("YQL query result error {!s}".format(error))
+
+                    time.sleep(self._delay)
+                except:
+                    logger.exception("YQL request excepted")
+                    raise
+
+        raise AssertionError("YQL request attempts has exhausted {} {}".format(self._attempts, query))
+
+
+    def request_expired_stat(self, aggregate_table, expired_threshold, timeout=None):
+        """
+        Send request to aggregate table to find volume of expired space in couples
+        :param aggregate_table: the name of aggregation table
+        :param expired_threshold: a couples with expired volume above param are suspected
+        :return list of couples with expired date more then trigger
+        """
+
+        timestamp = int(time.time())
+        query = self.aggregate_query.format(table=aggregate_table, timestamp=timestamp, trigger=expired_threshold)
+
+        try:
+            r = self.send_request(query, timeout)
+            if r.rows_count == 0:
+                logger.error("empty result")
+                return []
+        except:
+            logger.exception("YT request excepted")
+            return []
+
+        valid_couples = []
+
+        for table in r.results:  # access to results blocks until they are ready
+            table.fetch_full_data()
+            for row in table.rows:
+                try:
+                    couple = int(row[0])
+                except ValueError:
+                    logger.exception("couple {!s} is invalid".format(row[0]))
```

Comment:
Done in https://github.com/yandex/mastermind/pull/311/commits/4732106cd00d7b48fb9a6db30b7cff4aef8db6df